content/departments/cloud/technical-docs/custom_domain.md

```diff
@@ -92,3 +92,139 @@ It is worth nothing it does not make us any less secure. Our origin server still
+
+Provide the following information:
+
+- Custom Hostname: `$OLD_DNS_NAME`
+- Minimum TLS version: `TLS 1.2`
+- Certificate type: `Provided by Cloudflare` (default)
+- SSL Certificate Authority: `Let's Encrypt`
+- Certificate validation method: `TXT Validation`
+- Wildcard: **DO NOT ENABLE** (default)
+- Custom origin server: `$NEW_DNS_NAME`
+- Origin SNI value: `Origin server name`
+
+Finally, click `Add Custom Hostname` to submit the form.
+
+#### Add neccessary DNS records for the old domain
+
+Open the operation dashboard of the instance, and pick up half-way through the custom domain setup process:
+
+```sh
+export CLOUDFLARE_CUSTOM_HOSTNAME_ID=$(curl -sSLf -H "Authorization: Bearer $CF_API_TOKEN" "https://api.cloudflare.com/client/v4/zones/$NEW_ZONE_ID/custom_hostnames" | jq -r '.result[] | select( .hostname == "'$OLD_DNS_NAME'") | .id')
+export CLOUDFLARE_ZONE_ID=$NEW_ZONE_ID
+export CLOUDFLARE_API_TOKEN=$(gcloud secrets versions access latest --secret CLOUDFLARE_API_TOKEN --project sourcegraph-secrets)
+export DCV_RECORD="$(curl -sSLf -H "Authorization: Bearer $CLOUDFLARE_API_TOKEN" "https://api.cloudflare.com/client/v4/zones/$CLOUDFLARE_ZONE_ID/custom_hostnames/$CLOUDFLARE_CUSTOM_HOSTNAME_ID" | jq -rM '.result.ssl.dcv_delegation_records[0]')"
+export OWNERSHIP_RECORD="$(curl -sSLf -H "Authorization: Bearer $CLOUDFLARE_API_TOKEN" "https://api.cloudflare.com/client/v4/zones/$CLOUDFLARE_ZONE_ID/custom_hostnames/$CLOUDFLARE_CUSTOM_HOSTNAME_ID" | jq -rM '.result.ownership_verification')"
+```
+
+Then, starting from editing the `.status` field in `config.yaml` and complete the rest of the process to create all the neccessary DNS records on the old domain
+
+You changes will look like this PR: https://github.com/sourcegraph/infrastructure/pull/4893
+
+#### Backfil resources to terraform
+
+```sh
+cd terraform/stacks/network
+```
+
+```sh
+terraform import 'cloudflare_custom_hostname.network_ingresscustomhostname_2963CBB2' "$NEW_ZONE_ID/$CLOUDFLARE_CUSTOM_HOSTNAME_ID"
+```
+
+```sh
+terraform state rm cloudflare_record.network_ingresspublicipdns_B5ED0649
+terraform import 'cloudflare_record.network_ingresspublicipdns_B5ED0649' "$NEW_ZONE_ID/$NEW_DNS_NAME_RECORD_ID"
+```
+
+> You will see `cloudflare_record.network_ingresspublicipdns_B5ED0649` must be replaced but that's okay.
+> It's an unfortunate side effect of the way we pass the full DNS name as `name` instead of just the prefix of the subdomain.
+> Terraform will simply re-create the record with our desired name.
+> No downtime is expected
+
+```sh
+terraform apply -auto-approve
+```
+
+#### Cut over
+
+Update the existing dns record as a CNAME record for the new domain.
+
+```sh
+flarectl dns update --zone $OLD_ZONE --id $OLD_DNS_NAME_RECORD_ID --proxy=false --type CNAME --name $SLUG --content $NEW_DNS_NAME
+```
+
+#### Backfill more terraform resources
+
+Continue on https://github.com/sourcegraph/infrastructure/pull/4893
```

Comment:
Does this mean "continue with your equivalent of" this PR?

Reply:
updated ;)